src/main/java/seedu/address/logic/parser/AddClaimCommandParser.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CASH;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddClaimCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.claim.Amount;
+import seedu.address.model.claim.Claim;
+import seedu.address.model.claim.Description;
+import seedu.address.model.commonvariables.Name;
+import seedu.address.model.commonvariables.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddClaimCommandParser implements Parser<AddClaimCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddClaimCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DESCRIPTION, PREFIX_CASH, PREFIX_NAME,
+                        PREFIX_PHONE, PREFIX_TAG);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_DESCRIPTION, PREFIX_CASH, PREFIX_NAME, PREFIX_PHONE)
+                || !argMultimap.getPreamble().isEmpty()) {
```

Comment:
Method looks a bit long and difficult to understand, maybe it can be broken down into smaller parts with more separate methods?